extensions/functions_datetime.yaml

```diff
@@ -265,3 +354,264 @@ scalar_functions:
           - name: y
             value: interval_year
         return: boolean
+  -
+    name: "assume_timezone"
+    description: >-
+      Convert local timestamp to UTC-relative timestamp_tz using given local time's timezone.
+
+      Timezone strings must be as defined by IANA timezone database (https://www.iana.org/time-zones).
+      Examples: "Pacific/Marquesas", "Etc/GMT+1".
+      If timezone is not present then UTC is assumed. If timezone is invalid an error is thrown.
+    impls:
+      - args:
+          - name: x
+            value: timestamp
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
+        return: timestamp_tz
+      - args:
+          - name: x
+            value: date
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
+        return: timestamp_tz
+  -
+    name: "cast_timezone"
+    description: >-
+      Convert UTC-relative timestamp_tz to local timestamp using given local time's timezone.
+
+      Timezone strings must be as defined by IANA timezone database (https://www.iana.org/time-zones).
+      Examples: "Pacific/Marquesas", "Etc/GMT+1".
+      If timezone is not present then UTC is assumed. If timezone is invalid an error is thrown.
+    impls:
+      - args:
+          - name: x
+            value: timestamp_tz
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
+        return: timestamp
```

Comment:
I would prefer to actually not offer this option until it's actually requested. In my experience local times without the zone information are not very useful and if you just have to know the wall time to show to the user there's `strftime`.

Reply:
It's pretty fundamental in order to add/subtract calendar units to a `timestamp_tz` though, or to convert from one timezone to another, so I don't really see this as a formatting thing at all. I don't know. Keep in mind though that a consumer doesn't *have* to implement everything in Substrait core. If it's not implemented and doesn't end up being used much or at all there's no harm done IMO.